tool/tsh/kube_proxy.go

```diff
@@ -377,16 +449,64 @@ func kubeCertFromKey(key *client.Key, kubeCluster string) (tls.Certificate, erro
 	return cert, trace.Wrap(err)
 }
 
+// getCertReissuer returns a function that can reissue with MFA user certificate for accessing kubernetes cluster.
+// If required it performs relogin procedure.
+func (k *kubeLocalProxy) getCertReissuer(tc *client.TeleportClient) func(ctx context.Context, teleportCluster, kubeCluster string) (tls.Certificate, error) {
+	return func(ctx context.Context, teleportCluster, kubeCluster string) (tls.Certificate, error) {
+		var proxy *client.ProxyClient
+		var currentContext string
+		attempt := 0
+		err := client.RetryWithRelogin(ctx, tc, func() error {
+			attempt++
+			var err error
+
+			proxy, err = tc.ConnectToProxy(ctx)
+			if err != nil {
+				// We save user's current context in case there was an error and relogin procedure can happen,
+				// which will delete our ephemeral kubeconfig and we'll need to recreate it.
+				cfg, err := kubeconfig.Load(k.KubeConfigPath())
+				if err != nil {
+					log.WithError(err).Debugf("could not load ephemeral kubeconfig at %q", k.KubeConfigPath())
+				}
+
+				currentContext = cfg.CurrentContext
+			}
+			return trace.Wrap(err)
+		})
+		if err != nil {
+			return tls.Certificate{}, trace.Wrap(err)
+		}
+		defer proxy.Close()
+
+		if attempt == 2 {
```

Comment:
What's the reasoning behind `attempt==2`?

Reply:
That's how we track that relogin happened, since `attempt` could be 2 only if our callback has run second time, because successful relogin happened.

Reply:
To simplify the logic, we can always read and write the kubeconfig even if the user's certificate is valid

Reply:
Done.